slack.js

```diff
@@ -0,0 +1,266 @@
+//------------------------------------------------------------------------------
+// Bot / Slack related helpers
+//------------------------------------------------------------------------------
+var Botkit = require('botkit');
```

Comment:
Long term, I agree. But for this pull request I'd prefer to hold off on that work. 

I don't have the time to rewrite every command in the league to use this new paradigm, so I'm hoping I can just fix the ones that are broken by the new config format and then we can merge with master.  After that we can more slowly port over the rest of the commands and once they are ported, then we can fully internalize it.  

And until we fully internalize it, some of the commands in chesster.js and some in slack.js will need access to things, so we'll end up with a less ideal interface for using slack.js.  But we can fix that when we finish the porting work. 

If you're keen on getting it all ported over and are available to help, then I'm happy to start moving it that way.